content/install/integrations/stash.md

```diff
@@ -107,9 +107,15 @@ Drone users `git+https` to clone repositories, however, Bitbucket Server does no
 
 # Registration
 
-You must register your application with Bitbucket Server in order to generate a consumer key. Navigate to your account settings and choose Applications from the menu, and click Register new application. Now copy & paste the text value from `/etc/bitbucket/key.pub` into the `Public Key` in the incoming link part of the application registration.
+You must register your application with Bitbucket Server in order to generate a consumer key.
 
-Please use http://drone.mycompany.com/authorize as the Authorization callback URL.
+Navigate to your account settings (e.g. http://stash.mycompany.com/admin) and choose "Application Links" from the menu on the left.
+
+Enter the URL of your Drone server (e.g. http://drone.mycompany.com) and click "Create new link". Bitbucket server will warn you that no response was received from the URL you entered. Ignore this and click "Continue".
+
+Give the application a name (e.g. "Drone CI") and click "Continue".
+
+Edit the application you just created. Click on "Incoming Authentication". In the "Consumer Key" field enter the value of your `DRONE_STASH_CONSUMER_KEY` environment variable. In the "Consumer Name" field enter a suitable name, e.g. "Drone CI" (the exact value doesn't really matter). In the "Public Key" field enter the public key that you generated above (either the value of `DRONE_STASH_CONSUMER_RSA_STRING` or the contents of the file `DRONE_STASH_CONSUMER_RSA`). In the "Consumer Callback URL" enter the authorize URL, e.g. `http://drone.mycompany.com/authorize`. Click "Close".
```

Comment:
I'm getting an error  about the consumer key. 

Response Body: oauth_problem=consumer_key_unknown

I don't think we followed these instructions verbatim though - after re-reading the text I see that you're supposed to save the link, and then edit it again. 

It's also confusing that there's  a consumer key and a bunch of fields on the first pop up config dialog, and one field has "Consumer Key".   I think what we did wrong is add the key there,  then also on the next page. (Waiting for our bitbucket admin to come in and I can re-try and verify).   

It might be clearer to explicitly say what to ignore as was mentioned with the paragraph about no response from the app link. 

1. enter app link
   * ignore warning
2. Next dialog:  enter App Name
    * ignore all other fields
    * here, I don't think you can just click continue,  it gives us a warning unless we have "Create Incoming Link" selected.   So I guess you click that and then continue. 
3. On the next screen there's a consumer key field, so enter that.
    * put the public key in you generated earlier. 
    * callback url

Another thing that is confusing is after reading some other documentation you already have the authorize http link in mind.   We entered that link as the application link on the first step.   Maybe a caffein problem, maybe worth mentioning.


This is a screenshot of what I was talking about if you don't click the checkbox "Create Incoming Link". 

![image](https://user-images.githubusercontent.com/723989/28925011-d893d280-7820-11e7-9e28-4b394b272067.png)

I hope this feedback helps. 

If my hunch is wrong on the consumer key issue maybe you could correct me. 

Thanks!